### ISSUE TITLE
panic experienced when using vpn?

### ISSUE THREAD

**Issue by jonathanstrong** [SergejJurecko/mio_httpc#37]
flagging so you are aware, unfortunately have not identified what the root issue is here. I have a program using mio_httpc that has been chugging along fine for quite some time, however today it experienced a crash on startup, which traced back to this line in the v0.8.x codebase:

https://github.com/SergejJurecko/mio_httpc/blob/d29d56fad7eeabd547e62dbd4cd411f3829edc4d/src/resolve/mod.rs#L18

I started using a VPN for work in the past few days and when I disconnected the problem went away. I don't know enough about DNS to understand why that is, but figured I would flag as the code obviously did not contemplate it was possible to fail there. 

here is the backtrace:

```
thread '<unnamed>' panicked at 'called `Result::unwrap()` on an `Err` value: UnexpectedEOF', ~/.cargo/registry/src/github.com-1ecc6299db9ec823/mio_httpc-0.8.9/src/resolve/mod.rs:18:37
stack backtrace:
   0:     0x55ad00671b7c - std::backtrace_rs::backtrace::libunwind::trace::h2ab374bc2a3b7023
                               at /rustc/9bb77da74dac4768489127d21e32db19b59ada5b/library/std/src/../../backtrace/src/backtrace/libunwind.rs:90:5
   1:     0x55ad00671b7c - std::backtrace_rs::backtrace::trace_unsynchronized::h128cb5178b04dc46
                               at /rustc/9bb77da74dac4768489127d21e32db19b59ada5b/library/std/src/../../backtrace/src/backtrace/mod.rs:66:5
   2:     0x55ad00671b7c - std::sys_common::backtrace::_print_fmt::h5344f9eefca2041f
                               at /rustc/9bb77da74dac4768489127d21e32db19b59ada5b/library/std/src/sys_common/backtrace.rs:67:5
   3:     0x55ad00671b7c - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::h213003bc5c7acf04
                               at /rustc/9bb77da74dac4768489127d21e32db19b59ada5b/library/std/src/sys_common/backtrace.rs:46:22
   4:     0x55ad0069a88c - core::fmt::write::h78bf85fc3e93663f
                               at /rustc/9bb77da74dac4768489127d21e32db19b59ada5b/library/core/src/fmt/mod.rs:1126:17
   5:     0x55ad0066a2b5 - std::io::Write::write_fmt::he619515c888f21a5
                               at /rustc/9bb77da74dac4768489127d21e32db19b59ada5b/library/std/src/io/mod.rs:1667:15
   6:     0x55ad00673610 - std::sys_common::backtrace::_print::hf706674f77848203
                               at /rustc/9bb77da74dac4768489127d21e32db19b59ada5b/library/std/src/sys_common/backtrace.rs:49:5
   7:     0x55ad00673610 - std::sys_common::backtrace::print::hf0b6c7a88804ec56
                               at /rustc/9bb77da74dac4768489127d21e32db19b59ada5b/library/std/src/sys_common/backtrace.rs:36:9
   8:     0x55ad00673610 - std::panicking::default_hook::{{closure}}::h2dde766cd83a333a
                               at /rustc/9bb77da74dac4768489127d21e32db19b59ada5b/library/std/src/panicking.rs:210:50
   9:     0x55ad006731c7 - std::panicking::default_hook::h501e3b2e134eb149
                               at /rustc/9bb77da74dac4768489127d21e32db19b59ada5b/library/std/src/panicking.rs:227:9
  10:     0x55ad00673e34 - std::panicking::rust_panic_with_hook::hc09e869c4cf00885
                               at /rustc/9bb77da74dac4768489127d21e32db19b59ada5b/library/std/src/panicking.rs:624:17
  11:     0x55ad006738e0 - std::panicking::begin_panic_handler::{{closure}}::hc2c6d70142458fc8
                               at /rustc/9bb77da74dac4768489127d21e32db19b59ada5b/library/std/src/panicking.rs:521:13
  12:     0x55ad00672024 - std::sys_common::backtrace::__rust_end_short_backtrace::had58f7c459a1cd6e
                               at /rustc/9bb77da74dac4768489127d21e32db19b59ada5b/library/std/src/sys_common/backtrace.rs:141:18
  13:     0x55ad00673849 - rust_begin_unwind
                               at /rustc/9bb77da74dac4768489127d21e32db19b59ada5b/library/std/src/panicking.rs:517:5
  14:     0x55ad00698381 - core::panicking::panic_fmt::hf443e5eeb6cc9eab
                               at /rustc/9bb77da74dac4768489127d21e32db19b59ada5b/library/core/src/panicking.rs:96:14
  15:     0x55ad00698633 - core::result::unwrap_failed::h885d3f7beb571353
                               at /rustc/9bb77da74dac4768489127d21e32db19b59ada5b/library/core/src/result.rs:1617:5
  16:     0x55ad00112daf - mio_httpc::resolve::dns_parse::he24239990e18e59e
  17:     0x55ad000f70a2 - mio_httpc::connection::ConTable::event_send::h3259495a5535a252
  18:     0x55ad0012409e - mio_httpc::httpc::HttpcImpl::call_send::hc9c3473a10bf6a8d
  19:     0x55ad000ef43c - mio_httpc::api::simple_call::SimpleCall::perform::h5574351afdae176d
  20:     0x55ad000446b1 - [redacted]
  21:     0x55acffffb1c2 - [redacted]
  22:     0x55ad00017fcf - std::sys_common::backtrace::__rust_begin_short_backtrace::hae0f040f8475adec
  23:     0x55ad0000fd00 - core::ops::function::FnOnce::call_once{{vtable.shim}}::h377c60db722ec711
  24:     0x55ad0067ab63 - <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once::h59eef3b9c8a82350
                               at /rustc/9bb77da74dac4768489127d21e32db19b59ada5b/library/alloc/src/boxed.rs:1636:9
  25:     0x55ad0067ab63 - <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once::hb5bbe017c347469c
                               at /rustc/9bb77da74dac4768489127d21e32db19b59ada5b/library/alloc/src/boxed.rs:1636:9
  26:     0x55ad0067ab63 - std::sys::unix::thread::Thread::new::thread_start::h62931528f61e35f5
                               at /rustc/9bb77da74dac4768489127d21e32db19b59ada5b/library/std/src/sys/unix/thread.rs:106:17
  27:     0x7f8e45eff609 - start_thread
                               at /build/glibc-YbNSs7/glibc-2.31/nptl/pthread_create.c:477:8
  28:     0x7f8e45cba293 - clone
                               at /build/glibc-YbNSs7/glibc-2.31/misc/../sysdeps/unix/sysv/linux/x86_64/clone.S:95
  29:                0x0 - <unknown>
```

**Comment by SergejJurecko** [comment]
This is fixed in latest version

**Comment by jonathanstrong** [comment]
> This is fixed in latest version

simple/easy to backport to v0.8.x, or no? I am happy to submit PR if you can point me in the right direction.

**Comment by SergejJurecko** [comment]
Ok I've backported fixes from 0.9 to 0.8 and published 0.8.10

**Comment by jonathanstrong** [comment]
awesome! thank you!
